src/handlers/meta_transaction_handlers.ts

```diff
@@ -237,3 +373,129 @@ function parseV1RequestParams(req: express.Request): MetaTransactionV1QuoteReque
         priceImpactProtectionPercentage,
     };
 }
+
+/**
+ * Parse meta-transaction v2 quote and price params. The function calls `parseV1RequestParams` to parse
+ * shared params with v1 and then parses the fee configs param introduced in v2.
+ */
+function parseV2RequestParams(req: express.Request): MetaTransactionV2QuoteRequestParams {
+    const sharedV1RequestParams = parseV1RequestParams(req);
+    const parsedFeeConfigs = _parseFeeConfigs(req);
+
+    return {
+        ...sharedV1RequestParams,
+        feeConfigs: parsedFeeConfigs,
+    };
+}
+
+/**
+ * Parse the fee config param.
+ */
+function _parseFeeConfigs(req: express.Request): GaslessFeeConfigs | undefined {
+    let parsedFeeConfigs: GaslessFeeConfigs | undefined;
+
+    if (req.query.feeConfigs) {
+        const feeConfigs = req.query.feeConfigs as ParsedQs;
+        parsedFeeConfigs = {};
+
+        // Parse the integrator fee config
+        if (feeConfigs.integrator) {
+            const integratorFee: ParsedQs = feeConfigs.integrator as ParsedQs;
+
+            if (integratorFee.kind !== 'volume') {
```

Comment:
isn't volume the only kind for integratorFee?

Reply:
Yes for now. Here trying to add an extra validation here for the case that the caller might provide the wrong `kind` and provide error code.